### PR TITLE
CosmosContainerBuilder with public constructor

### LINKED ISSUE #599: CosmosContainerBuilder with public constructor
CosmosContainerBuilder constructor should be internal

### PR DESCRIPTION
**Pull request by ealsur**
## Description

CosmosContainerBuilder  had a public empty ctor that caused confusion while discovering APIs.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Closing issues

This PR closes #599

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Fluent/Settings/ContainerBuilder.cs [hunk 1]
```diff
@@ -18,13 +18,6 @@ public class ContainerBuilder : ContainerDefinition<ContainerBuilder>
         private UniqueKeyPolicy uniqueKeyPolicy;
         private ConflictResolutionPolicy conflictResolutionPolicy;
 
-        /// <summary>
-        /// Creates an instance for unit-testing
-        /// </summary>
-        public ContainerBuilder()
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Fluent/Settings/ContainerBuilder.cs]: Can we leave this as protected so it can still be mocked?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Fluent/Settings/ContainerBuilder.cs]: Done

**Comment by bartelink** [comment]
those azp command sure generate noise for watchers ;)

**Comment by ealsur** [comment]
@bartelink Sorry for the inconvenience, we are migrating to in-code definitions, and the opened PR seem to be stuck in the middle

**Comment by bartelink** [comment]
ha, it's not an actual problem, I was only musing to myself - hopefully github will at some point filter such meta-comments (perhaps octobox et al already do?)

**Pull request merged**